src/OrleansTestKit/Storage/StorageManager.cs

```diff
@@ -1,32 +1,57 @@
 ﻿using System;
+using System.Collections.Generic;
+using Moq;
 using Orleans.Core;
+using Orleans.Runtime;
 
 namespace Orleans.TestKit.Storage
 {
     public sealed class StorageManager
     {
         private readonly TestKitOptions _options;
 
-        private object _storage;
+        private readonly Dictionary<string, object> _storages = new Dictionary<string, object>();
 
         public StorageManager(TestKitOptions options) =>
             _options = options ?? throw new ArgumentNullException(nameof(options));
 
-        public IStorage<TState> GetStorage<TState>()
+        internal readonly Mock<IAttributeToFactoryMapper<PersistentStateAttribute>> stateAttributeFactoryMapperMock =
+            new Mock<IAttributeToFactoryMapper<PersistentStateAttribute>>();
+
+        public IStorage<TState> GetStorage<TState>() => GetStorage<TState>("Default");
```

Comment:
It's an edge case... but if a grain extends `Grain<TState>` _and_ has a second state using the facets/DI that just so happens to use the name "Default" then this implementation will return the same instance for both cases whereas Orleans will not. Is it possible to eliminate this potential overlap?

Reply:
I've reverted this so that test kit using the same method as orleans for naming the grain state i.e. the FullName of the grain type. 

This means the grain type has to be provided when looking up the grain state as you're not providing a state name. I normalised this across the board so a state name or a grain type is required for lookup.